xdsl/ir/affine/affine_map.py

```diff
@@ -235,6 +235,39 @@ def inverse_permutation(self) -> AffineMap | None:
             results=results,
         )
 
+    def inverse_and_broadcast_projected_permutation(self):
+        """
+        If `self` is a projected permutation, with possible constant 0 expression
+        results, returns the inverse permutation.
```

Comment:
Is calling this map an inverse standard terminology? It's not an inverse in the mathematical sense of an affine map

Reply:
True